src/Backend/Backend.ts

```diff
@@ -37,5 +37,5 @@ export interface Backend {
   compiler(): Compiler|undefined;
 
   // executor specs by being filled by impl
-  executor(): Executor|undefined;
+  executors(): Executor[]|undefined;
```

Comment:
Should it include `undefined`? As it's an array, we can replace the use case of `undefined` with an empty array.